dbt_platform_helper/commands/environment.py

```diff
@@ -572,50 +525,50 @@ def add_maintenance_page(
     env: str,
     services: List[Service],
     allowed_ips: tuple,
-    ip_filter: bool,
     template: str = "default",
 ):
     lb_client = session.client("elbv2")
     maintenance_page_content = get_maintenance_page_template(template)
+    bypass_value = "".join(random.choices(string.ascii_lowercase + string.digits, k=12))
 
-    for svc in services:
-        target_group_arn = find_target_group(app, env, svc.name)
+    for index, svc in enumerate(services):
+        target_group_arn = find_target_group(app, env, svc.name, session)
 
         # not all of an application's services are guaranteed to have been deployed to an environment
         if not target_group_arn:
             continue
 
-        if not ip_filter:
-            user_ip = get_public_ip()
-            allowed_ips = list(allowed_ips) + [user_ip]
-
+        user_ip = get_public_ip()
+        allowed_ips = list(allowed_ips) + [user_ip]
+        for ip_index, ip in enumerate(allowed_ips):
             create_header_rule(
                 lb_client,
                 listener_arn,
                 target_group_arn,
                 "X-Forwarded-For",
-                allowed_ips,
+                [ip],
                 "AllowedIps",
-            )
-        else:
-            bypass_value = "".join(random.choices(string.ascii_lowercase + string.digits, k=12))
-            create_header_rule(
-                lb_client,
-                listener_arn,
-                target_group_arn,
-                "Bypass-Key",
-                [bypass_value],
-                "BypassIpFilter",
+                (index + 1) * (ip_index + 100),
             )
 
-            click.secho(
-                f"\nUse a browser plugin to add `Bypass-Key` header with value {bypass_value} to your requests. For more detail, visit https://platform.readme.trade.gov.uk/ ",
-                fg="green",
-            )
+        create_header_rule(
+            lb_client,
+            listener_arn,
+            target_group_arn,
+            "Bypass-Key",
+            [bypass_value],
+            "BypassIpFilter",
+            index + 1,
+        )
+
+        click.secho(
+            f"\nUse a browser plugin to add `Bypass-Key` header with value {bypass_value} to your requests. For more detail, visit https://platform.readme.trade.gov.uk/ ",
+            fg="green",
+        )
 
     lb_client.create_rule(
         ListenerArn=listener_arn,
-        Priority=2,
+        Priority=20000,  # big number because we create multiple higher priority "AllowedIps" rules for each allowed ip for each service above.
```

Comment:
Could we increment a `priority` variable above and do the following here?
```suggestion
        Priority=priority + 1,  # big number because we create multiple higher priority "AllowedIps" rules for each allowed ip for each service above.
```